File: demo/common/src/main/java/com/serious/demo/TestRemoteRestService.java
package com.serious.demo;/*
 * @COPYRIGHT (C) 2023 Andreas Ernst
 *
 * All rights reserved
 */

import com.serious.service.Service;
import com.serious.service.ServiceInterface;
import org.springframework.web.bind.annotation.*;
import reactor.core.publisher.Mono;


/**
 * @author Andreas Ernst
 */
@ServiceInterface()
@RestController
public interface TestRemoteRestService extends Service {
    @GetMapping("/hello")
    String hello();

    @GetMapping("/passId/{id}")
    String passId(@PathVariable("id") String id);

    @GetMapping("/passIdReturnBody/{id}")
    @ResponseBody
    Foo passIdReturnBody(@PathVariable("id") String id);

    @PostMapping("/postBody")
    @ResponseBody
    Foo postBody(@RequestBody Foo foo);

    @GetMapping("/passParam")
    String passParam(@RequestParam("id") String id);

    @PostMapping("/postBodyMono")
    @ResponseBody
    Mono<Foo> postBodyMono(@RequestBody Foo foo);
}

File: demo/common/src/main/java/com/serious/demo/TestRemoteRestService.kt
package com.serious.demo

import com.serious.service.Service
import com.serious.service.ServiceInterface
import org.springframework.web.bind.annotation.*
import reactor.core.publisher.Mono

/*
* @COPYRIGHT (C) 2023 Andreas Ernst
*
* All rights reserved
*/ /**
 * @author Andreas Ernst
 */
@ServiceInterface
@RestController
interface TestRemoteRestService : Service {
    @PostMapping("/throw")
    @ResponseBody
    @Throws(RuntimeException::class)
    fun throwException(@RequestBody id: String): Void

    @PostMapping("/throwFatal")
    @ResponseBody
    fun throwFatalException(@RequestBody id: String): Void

    @GetMapping("/hello")
    fun hello(): String

    @GetMapping("/passId/{id}")
    fun passId(@PathVariable("id") id: String): String

    @GetMapping("/passIdReturnBody/{id}")
    @ResponseBody
    fun passIdReturnBody(@PathVariable("id") id: String): Foo

    @PostMapping("/postBody")
    @ResponseBody
    fun postBody(@RequestBody foo: Foo): Foo

    @GetMapping("/passParam")
    fun passParam(@RequestParam("id") id: String): String

    @PostMapping("/postBodyMono")
    @ResponseBody
    fun postBodyMono(@RequestBody foo: Foo): Mono<Foo>
}
